src/coreclr/src/vm/bundle.cpp

```diff
@@ -30,25 +31,25 @@ const SString &BundleFileLocation::Path() const
     return Bundle::AppBundle->Path();
 }
 
-Bundle::Bundle(LPCWSTR bundlePath, BundleProbe *probe)
+Bundle::Bundle(LPCSTR bundlePath, BundleProbe *probe)
 {
     STANDARD_VM_CONTRACT;
 
     _ASSERTE(probe != nullptr);
 
-    m_path.Set(bundlePath);
+    m_path.SetUTF8(bundlePath);
     m_probe = probe;
 
     // The bundle-base path is the directory containing the single-file bundle.
     // When the Probe() function searches within the bundle, it masks out the basePath from the assembly-path (if found).
 
-    LPCWSTR pos = wcsrchr(bundlePath, DIRECTORY_SEPARATOR_CHAR_W);
+    LPCSTR pos = strrchr(bundlePath, DIRECTORY_SEPARATOR_CHAR_A);
     _ASSERTE(pos != nullptr);
-    size_t baseLen = pos - bundlePath + 1; // Include DIRECTORY_SEPARATOR_CHAR_W in m_basePath
-    m_basePath.Set(bundlePath, (COUNT_T)baseLen);
+    size_t baseLen = pos - bundlePath + 1; // Include DIRECTORY_SEPARATOR_CHAR_A in m_basePath
+    m_basePath.SetUTF8(bundlePath, (COUNT_T)baseLen);
 }
 
-BundleFileLocation Bundle::Probe(LPCWSTR path, bool pathIsBundleRelative) const
+BundleFileLocation Bundle::Probe(LPCSTR path, bool pathIsBundleRelative) const
```

Comment:
I would keep just the `LPCWSTR` version. You really need just the change in the constructor above. The `SString` has `operator const WCHAR *` defined so that it performs conversion to unicode if the string is not unicode already. That conversion is "in place", so the `SString` contains the unicode version after that.
The operator is implicit, so the call to `wcsncmp(m_basePath, path, baseLen)` below would do the right thing.
That would also avoid conversions at each call to `ProbeAppBundle` where the string passed in is unicode (the calls in AssemblyBinder.cpp).

Reply:
In the end, we need `LPCSTR path` for `m_probe(path, &loc.Offset, &loc.Size)` call below. Before this change, we were doing this conversion in the host, now we are doing it in CoreCLR, from that sense. Overall it is slightly expensive (one extra conversion) on Windows. I agree with rest of your comment; changed this to `SString path` and removed `ProbeAppBundle(SString)` overload.